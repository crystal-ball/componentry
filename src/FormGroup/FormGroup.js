import { staticComponent } from '../utils/static-component-builder'

/**
 * [FormGroup component 📝](https://componentry.design/components/form-group)
 */
export const FormGroup = staticComponent('FormGroup', {
  componentCx: 'form-group',
})
