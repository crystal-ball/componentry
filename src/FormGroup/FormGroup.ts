/* eslint-disable @typescript-eslint/no-empty-interface */

import { ComponentBaseProps } from '../utils/types'
import { staticComponent } from '../utils/static-component-builder'

export interface FormGroupProps extends ComponentBaseProps<'div'> {}

/**
 * [FormGroup component üìù](https://componentry.design/components/form-group)
 * @experimental
 */
export const FormGroup = staticComponent('FormGroup')
