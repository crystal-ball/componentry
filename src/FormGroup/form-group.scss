/* componentry/src/FormGroup/form-group */

.ðŸ…²-form-group {
  margin-bottom: $form-group-margin-bottom;
}
