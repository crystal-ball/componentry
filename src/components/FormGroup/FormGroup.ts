import { type ComponentBaseProps } from '../../utils/base-types'
import { staticComponent } from '../../utils/static-component-builder'

export interface FormGroupProps extends ComponentBaseProps<'div'> {}

/**
 * [FormGroup component üìù](https://componentry.design/components/form-group)
 * @experimental
 */
export const FormGroup = staticComponent('FormGroup')
