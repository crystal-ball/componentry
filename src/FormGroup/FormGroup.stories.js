/* global module */

import React from 'react'
import { storiesOf } from '@storybook/react'

import FormGroup from './FormGroup'

storiesOf('FormGroup', module).add('<FormGroup />', () => (
  <FormGroup>Form stuff</FormGroup>
))
