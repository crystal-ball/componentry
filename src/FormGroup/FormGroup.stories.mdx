import { DocsTitle, Meta } from '@storybook/addon-docs/blocks'
import { FormGroup } from './FormGroup'

<Meta title='Components/FormGroup' component={FormGroup} />

<DocsTitle title='Form Groups' experimental />

<FormGroup>Form groups provide the wrapper class for spacing in forms</FormGroup>
