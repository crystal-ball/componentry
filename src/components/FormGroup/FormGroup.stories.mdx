import { Meta, Story } from '@storybook/addon-docs'

import { Input } from '../Input/Input'
import { FormGroup } from './FormGroup'

<Meta title='Components/FormGroup' component={FormGroup} />

# FormGroup

Form groups provide a control point for standardizing spacing within forms.

<Story name='FormGroup'>
  <div className='w-1/2'>
    <FormGroup>
      <Input>
        <Input.Label>Name</Input.Label>
        <Input.Field placeholder='' />
      </Input>
    </FormGroup>
    <FormGroup>
      <Input>
        <Input.Label>Email</Input.Label>
        <Input.Field placeholder='' />
      </Input>
    </FormGroup>
  </div>
</Story>
