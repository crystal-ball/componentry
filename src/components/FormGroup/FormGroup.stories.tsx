import type { Meta, StoryObj } from '@storybook/react'

import { Input } from '../Input/Input'
import { FormGroup } from './FormGroup'

const meta: Meta<typeof FormGroup> = {
  component: FormGroup,
}

export default meta
type Story = StoryObj<typeof FormGroup>

export const Primary: Story = {
  render: () => (
    <div className='w-1/2'>
      {/* @ts-expect-error -- experimental component */}
      <FormGroup>
        <Input>
          <Input.Label>Name</Input.Label>
          <Input.Field placeholder='' />
        </Input>
      </FormGroup>
      {/* @ts-expect-error -- experimental component */}
      <FormGroup>
        <Input>
          <Input.Label>Email</Input.Label>
          <Input.Field placeholder='' />
        </Input>
      </FormGroup>
    </div>
  ),
}
