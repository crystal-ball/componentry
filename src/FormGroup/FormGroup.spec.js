import React from 'react'
import { render } from '@testing-library/react'

import elementTests from '../../test/element-tests'
import FormGroup from './FormGroup'

describe('<FormGroup />', () => {
  // Basic library element test suite
  elementTests(FormGroup)
})

// Snapshots
// ---------------------------------------------------------------------------
describe('<FormGroup /> snapshots', () => {
  it('renders correctly', () => {
    const { container } = render(<FormGroup />)
    expect(container.firstChild).toMatchSnapshot()
  })
})
