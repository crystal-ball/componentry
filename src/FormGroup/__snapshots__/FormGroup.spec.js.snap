// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<FormGroup /> snapshots renders correctly 1`] = `
<div
  class="form-group"
/>
`;
